azure-configurationservice/tests/test_azure_configuration_client.py

```diff
@@ -0,0 +1,601 @@
+        )
+        assert (
+            created_kv.etag is not None
+            and created_kv.last_modified is not None
+            and created_kv.locked is False
+        )
+
+    def test_add_existing_configuration_setting(self):
+        kv = ConfigurationSetting()
+        compare_kv = self.test_data.label1_data[0]
+        kv.key = compare_kv.key
+        kv.label = compare_kv.label
+        kv.value = compare_kv.value
+        kv.content_type = compare_kv.content_type
+
+        with pytest.raises(CloudError):
+            self.app_config_client.add_configuration_setting(kv)
+
+    def test_add_configuration_setting_no_label(self):
+        kv = ConfigurationSetting()
+        kv.key = "unit_test_key_add" + self.test_data.key_uuid
+        # kv.label is None by default
+        kv.value = "test value"
+        kv.content_type = "test content type"
+        kv.tags = {"tag1": "tag1", "tag2": "tag2"}
+        kv.etag = "not needed"  # this etag will not be processed.
+        created_kv = self.app_config_client.add_configuration_setting(kv)
+        self.app_config_client.delete_configuration_setting(kv.key, kv.label)
+        assert (
+            kv.value == kv.value
+            and created_kv.content_type == kv.content_type
+            and created_kv.tags == kv.tags
+        )
+        assert (
+            created_kv.etag is not None
+            and created_kv.last_modified is not None
+            and created_kv.locked is False
+            and created_kv.label is None
+        )
+
+    def test_add_existing_configuration_setting_no_label(self):
+        kv = ConfigurationSetting()
+        compare_kv = self.test_data.no_label_data[0]
+        kv.key = compare_kv.key
+        kv.value = compare_kv.value
+        kv.content_type = compare_kv.content_type
+
+        with pytest.raises(CloudError):
+            self.app_config_client.add_configuration_setting(kv)
+
+    # method: set_configuration_setting
+    def test_set_existing_configuration_setting_no_label_etag(self):
+        sample_kv = copy(self.test_data.no_label_data[-1])
+
+        # create a new key value into AzConfig service
+        sample_kv.key = "unit_test_key_set_" + self.test_data.key_uuid
+        to_set_kv = self.app_config_client.add_configuration_setting(sample_kv)
+        to_set_kv.value = to_set_kv.value + "a"
+        to_set_kv.tags = {"a": "b", "c": "d"}
+        set_kv = self.app_config_client.set_configuration_setting(to_set_kv)
+        # remove the new key value from the service
+        self.app_config_client.delete_configuration_setting(set_kv.key)
+        assert (
+            to_set_kv.key == set_kv.key
+            and None is set_kv.label
+            and to_set_kv.value == set_kv.value
+            and to_set_kv.content_type == set_kv.content_type
+            and to_set_kv.tags == set_kv.tags
+            and to_set_kv.etag != set_kv.etag
+        )
+
+    def test_set_existing_configuration_setting_label_etag(self):
```

Comment:
It is across all of your tests.
See `test_set_existing_configuration_setting_no_label_etag` vs `test_set_existing_configuration_setting_label_etag`.do we really need both? is there a way the SDK functionality is affected by making the distinction between label and no label?
I will stop these types of comments across the tests, but it will be good to make the exercise and reduce the number of tests to the ones that are actually validating the SDK and not the service